### ISSUE TITLE
Observing changes from two tables and merging into a single array

### ISSUE THREAD

**Issue by lukasb** [dexie/Dexie.js#2090]
I'm working on an app that has a main data table and a table of queued updates. I have a single in-memory array of items, which combines entries from the main table and the queued updates in sort of a complicated way (pull everything from main table, if an item has an entry in the queued updates table replace with that, unless the update is a deletion, in which case remove.)

As far as I can tell, I can't use liveQuery, because I don't want to have a second copy of my main table in memory. However, I know use of Dexie.Observable is discouraged. Any recommendations?

**Comment by dfahlander** [comment]
I still think you should be using liveQuery and use the `{cache: 'immutable'}` option in [Dexie constructor](https://dexie.org/docs/Dexie/Dexie). Within the callback of liveQuery you could query both tables according to your algorithm and return a final result.

By using `{cache: 'immutable'}`, the cache will not have to maintain a deeply cloned copy of the queries.

In the end, your RAM will maintain both the intermediate results and the derived result (the return value from your liveQuery callback), but this only regards to the array of references to objects - the objects themselves would not be duplicated unless you clone them when computing the result. Probably the arrays of references would take negligible RAM.

If your goal would be to heavily optimize RAM even more, you could use the option `{ cache: 'disabled' }` and the intermediate query results won't be cached at all - only the result of your querier callback.

**Comment by dfahlander** [comment]
> Thanks. The querier I provide to useLiveQuery can only observe a single table though, right? So I'll need two useLiveQuery statements, each one will look at both tables and modify my in-memory array. The querier will basically be equivalent for both.

No, there's no such limitation in the querier. You can query multiple tables in loops or however you want, even possible to query multiple databases. The only restrictions are that your queries are read-only (no add, put or update) and that you don't await non-dexie promises (such as fetch() etc).
